### DISCUSSION TITLE
Is it possible for a user to disable specific hooks

### DISCUSSION THREAD

**Discussion by CircleCode** [captainhook-git/captainhook#214] [Q&A] [answered]
as stated in [the documentation](https://captainhookphp.github.io/captainhook/configure.html#settings-overwrite), it is possible to overwrite the config section, but this does not seem to allow a user to disable a hook through the `captainhook.config.json` file.

Is there a way for a single user to disable a given hook without chnaging the main config file, shared in git?

**Comment by sebastianfeldmann** [comment, marked as answer]
Disabling a hook via local `config.json` is currently not possible. It was a conscious decision not to do so because the main idea behind the Cap'n was to manage all hooks via git for the whole team and having everybody executing the same hooks.

You have multiple options.

- Skip the hook you want to disable manually `git -n` or `CAPTAINHOOK_SKIP_HOOKS=1 git ...` (you could create an alias for it)
- Create two configs
- Play around with a sub config

#### Two configs

1. Copy your config to `captainhook.nohook-x.json`
2. And install it for the special team members like this `captainhook install --configuration=captainhook.nohook-x.json -f`

With this everybody can just proceed as normal and the ones who want to disable the actions can use the special config.

#### Sub config

1. Create a config `A` without any actions for the hook in question
2. Create a second config file `B` with just the actions for the hook in question
3. And now the playing around part. Don't include it in config `A` but for everybody who wants to execute the actions create a local `config.json` including the `B` config

This has the advantage that you do not need to copy paste the config but it's a bit more complicated in terms of making everybody execute the same hooks.

**Reply by CircleCode** [reply]
thanks for the very detailed answer :-)

I suspected this was on purpose, and I can only adhere to this approach.

I thought about the subconfig and found it cumbersome, but did not think about the easy Two configs approach!
